test/utils/cludformation.js

```diff
@@ -57,9 +59,50 @@ function listStacks(status) {
   return awsRequest('CloudFormation', 'listStacks', params);
 }
 
+async function doesStackWithNameAndStatusExists(name, status) {
+  try {
+    const describeStacksResponse = await awsRequest('CloudFormation', 'describeStacks', {
+      StackName: name,
+    });
+    if (describeStacksResponse.Stacks[0].StackStatus === status) {
+      return true;
+    }
+    return false;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function getStackOutputMap(name) {
+  const describeStackResponse = await awsRequest('CloudFormation', 'describeStacks', {
+    StackName: name,
+  });
+
+  return describeStackResponse.Stacks[0].Outputs.reduce((map, output) => {
+    map[output.OutputKey] = output.OutputValue;
+    return map;
+  }, {});
```

Comment:
style nit: `reduce` is controversial for such processing. Wouldn't it be more readable to have it as:

```javascript
const outputsMap = new Map();
for (const { OutputKey: key, OutputValue: value } of describeStackResponse.Stacks[0].Outputs) {
  outputsMap.set(key, value);
}
return outputsMap;
```

I've also used _Map_, as in normal processing is more convienient (and less error-prone), still if for some reason, we need to transport it as JSON string somewhere, the going with plain object is definitely way to go.

Reply:
I like the proposed solution - I think that it's in fact a bit more obvious at first sight what is going on there.